backend/tests/test_middleware.py

```
@@ -0,0 +1,612 @@
+        mock_request.url.path = "/form-submit"
+
+        with patch.object(middleware, '_validate_csrf_token', return_value=False):
+            with pytest.raises(HTTPException) as exc_info:
+                await middleware.dispatch(mock_request, AsyncMock())
+
+            assert exc_info.value.status_code == status.HTTP_403_FORBIDDEN
+            assert "CSRF token validation failed" in str(exc_info.value.detail)
+
+    def test_validate_csrf_token_success(self, middleware, mock_request):
+        """Test successful CSRF token validation."""
+        token = "valid_csrf_token"
```

Ruff (0.13.1)

Possible hardcoded password assigned to: "token"

(S105)